caput/memh5.py

```diff
@@ -2146,7 +2146,8 @@ def _map_json(value):
         def _convert_datetime(v):
             if isinstance(v, datetime.datetime):
                 return v.isoformat()
-            raise TypeError("Can not JSON serialise object of type %s" % repr(type(v)))
+            # Let the default method raise the TypeError
+            return json.JSONEncoder.default(json.JSONEncoder, v)
```

Comment:
I'm pretty sure `json.JSONEncode.default` is an instance method, so this only works because the particular implementation of it doesn't use the `self` argument (or doesn't use it to access instance attributes). Seems pretty risky to rely on that implementation not changing.

If we do want to call the base implementation I would just do a very quick subclass of `JSONEncoder` that moves the `_convert_datetime` implementation into there.